fe/fe-core/src/main/java/com/starrocks/http/rest/TableQueryPlanAction.java

```diff
@@ -142,8 +144,24 @@ protected void executeWithoutPassword(BaseRequest request, BaseResponse response
                 throw new StarRocksHttpException(HttpResponseStatus.BAD_REQUEST,
                         "POST body must contains [sql] root object");
             }
-            LOG.info("receive SQL statement [{}] from external service [ user [{}]] for database [{}] table [{}]",
-                    sql, ConnectContext.get().getCurrentUserIdentity(), dbName, tableName);
+
+            String warehouseName = WarehouseManager.DEFAULT_WAREHOUSE_NAME;
+            if (request.getRequest().headers().contains(WAREHOUSE_KEY)) {
+                warehouseName = request.getRequest().headers().get(WAREHOUSE_KEY);
+                Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouseAllowNull(warehouseName);
+                long warehouseId;
+                if (warehouse != null) {
+                    warehouseId = warehouse.getId();
+                } else {
+                    LOG.warn("The warehouse parameter [{}] is invalid, use default warehouse instead", warehouseName);
+                    warehouseId = WarehouseManager.DEFAULT_WAREHOUSE_ID;
```

Comment:
Shall report error, because the request can't be fulfilled in the requested warehouse. Don't overrule the request input.